application/api/controller.py

```diff
@@ -0,0 +1,462 @@
+        if isinstance(user_id, int):
+            items = BucketListItems.query.filter_by(
+                bucketlist_id=bucketlist_id).all()
+            if items:
+                item_list = []
+                for item in items:
+                    item_list.append({
+                        "id": item.id,
+                        "name": item.name,
+                        "message": "Bucketlist item obtained successfully.",
+                        "date_created": item.date_created,
+                        "done": item.done
+                    })
+                return make_response(
+                    jsonify({"Bucketlistitems":
+
+                             item_list}), 200)
+        else:
+            return make_response(
+                jsonify(
+                    {
+                        'message': user_id
+
+                    }
+                ), 401)
+
+
+class SingleBucketListItem(Resource):
+
+    def get(self, bucketlist_id, item_id):
+        token = request.headers.get('Authorization')
+        user_id = User.verify_token(token)
+        if isinstance(user_id, int):
+            item = BucketListItems.query.filter_by(
+                bucketlist_id=bucketlist_id, id=item_id).first()
+            if item:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "Bucketlist item obtained successfully.",
+                            'item_id': item.id,
+                            'name': item.name,
+                            'date_created': item.date_created,
+                            'date_modified': item.date_modified
+
+
+                        }
+                    ), 200)
+
+            return make_response(
+                jsonify(
+                    {
+                        'message': "Bucket list item with the id does not exist."
+
+                    }
+                ), 404)
+        return make_response(
+            jsonify(
+                {
+                    'message': user_id
+
+                }
+            ), 401)
+
+    def put(self, bucketlist_id, item_id):
+        token = request.headers.get('Authorization')
+        user_id = User.verify_token(token)
+        if isinstance(user_id, int):
+            bucketlistitem = BucketListItems.query.filter_by(
+                bucketlist_id=bucketlist_id, id=item_id).first()
+            if not bucketlistitem:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "Bucketlist item not found."
+
+                        }
+                    ), 404)
+
+            new_bucketlistitem_name = request.json["name"]
+            if not new_bucketlistitem_name:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "name field is missing."
+
+                        }
+                    ), 403)
```

Comment:
status code `400` would seem to be a better response considering the client has to make changes to the request for it to be successful